src/components/edit-movie/edit-movie.tsx

```diff
@@ -2,30 +2,68 @@ import classNames from 'classnames';
 import { AppButton } from 'components/app-button/app-button';
 import { Genres } from 'components/menu/menu';
 import { Modal } from 'components/modal/modal';
+import { useState } from 'react';
+import { useAppDispatch } from 'redux/hooks';
+import { addMovieAction, updateMovieAction } from 'redux/movies';
+import { Movie } from 'types/movie';
 import styles from './edit-movie.module.scss';
 
 const genres = [Genres.Comedy, Genres.Crime, Genres.Documentary, Genres.Horror];
 
+type MovieFormValues = Partial<Movie>;
+
 interface Props {
   title: string;
+  movie?: Movie;
   onClose: () => void;
+  action: 'add' | 'edit';
 }
 
-export const EditMovie = ({ title, onClose }: Props) => {
+export const EditMovie = ({ title, movie, onClose, action }: Props) => {
+  const dispatch = useAppDispatch();
+
+  const [movieData, setMovieData] = useState<MovieFormValues | undefined>(
+    movie,
+  );
+
   const handleSubmit = () => {
+    if (!movieData) {
+      return;
+    }
+    if (!isValidMovie(movieData)) {
+      return;
+    }
+    if (action === 'add') {
+      dispatch(addMovieAction(movieData));
+    }
+    if (movie?.id) {
+      dispatch(updateMovieAction({ ...movieData, id: movie.id }));
+    }
     onClose();
   };
   const handleReset = () => {
     onClose();
   };
 
+  const setData = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+    field: string,
+  ) => {
+    setMovieData((prevState) => ({
+      ...prevState,
+      [field]: e.target.value,
+    }));
+  };
+
   return (
     <Modal onClose={handleReset}>
       <div className={styles.title}>{title}</div>
       <div className={styles.content}>
         <div className={styles.textWrapper}>
           <label className={styles.label}>Title</label>
           <input
+            value={movieData?.title}
```

Comment:
Are there any errors in console related to passing `null` value for input?
Anyway, for `input` values I strongly advise to always set some default value, like `''` or `0` to prevent wrong behaviour or some JSX errors